app/services/query-service.js

```diff
@@ -108,13 +112,17 @@ const runFetchAnswerQuery = async ({ query, chatHistory, summariesMode, embeddin
       input: redactedQuery
     })
 
-    return response
+    const hallucinated = !validateResponseLinks(response, query)
+
+    return { response, hallucinated }
```

Comment:
I changed this format to to check every response and return if its hallucianeted, which is especially important for the summaries search as it, by the virtue of it being created so much later, will have much less information and thus is more likely to halluciante